### PR TITLE
Add memory monitor to GH execs

### PR DESCRIPTION
**Pull request by knelli2**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]
LGTM but needs a rebase to resolve conflicts. Please ping the core devs (maybe Nils Vu?) once you've rebased to get this merged since I'll be away

**Comment by knelli2** [comment]
@nilsvu Could you take a quick look please?

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
